### PR TITLE
Religion datums (ready for review)

### PR DESCRIPTION
**Pull request by ShiftyRail**
## Scope of this PR
Reworks the chaplain code to have religion datums rather than an endless list of `if`.

## Reasoning
The code should (hopefully) be easier to read and to maintain.
Programming snowflake interactions, more advanced mechanics, you name it, should be much easier (converting people to your religion, with in-game effects ? Why not)
Brings the codebase closer to the "standards".
It also fix very minor bugs, so minor that nobody bothered to report them yet.

## Todo
- [x] Adds the few religions that I've not typed yet
- [x] Removing dead code
- [x] Correcting all the dumb copypasting mistakes I did holy shit

All things necessary done, now let's discuss what could be done better.
For now, I think we are doing things purely for flavour, no real gameplay changes intended. 
- [x] Bible linked to religion ?

And maybe in another PR :
- [ ] `convert()` proc
- [ ] Spessmen having their own religion datum, and a religion datum having a reference towards its spiritual guide and its converts, so that admins can identify who follow a particular religion and do ~~fun~~ buses about it

Thanks to Shadowmech88 (and other people in the coding discord) for helping me with this.

### CODE REVIEW

**Comment by Knightkiller11** [comment]
Buddism seems to have the same keywords as hinduism

File: code/game/jobs/job/civilian_chaplain.dm [hunk 1]
```diff
@@ -34,11 +33,41 @@
...
+			new_religion = chap_religion.name // If nothing was typed
+		new_religion = lowertext(new_religion)
+
+		var/datum/job/J = H.mind.role_alt_title
+
+		for (var/R in typesof(/datum/religion))
+			var/datum/religion/rel = new R
+			var/halt = FALSE
+			for (var/key in rel.keys)
+				to_chat(world, "[key]")
```
Comment (MadmanMartian) [inline code/game/jobs/job/civilian_chaplain.dm]: Debug code left over. Panic.

**Comment by ShiftyRail** [comment]
Seems like Travis didn't like it
Probably something to do with the file tree (I did not include religions.dm)
Will fix it + the comments tommorow

Good night everyone

File: code/game/jobs/job/religions.dm [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+	name = "The Ultimate Question"
+	deity_name = "42"
+	bible_name = "The Hitchhiker's Guide to the Galaxy"
+	bible_type = /obj/item/weapon/storage/bible/booze
+	male_adept = "Hitchhiker"
+	female_adept = "Hitchhiker"
+	keys = list("coding", "coder", "coders", "coderbus")
+
+/datum/religion/spooky
+	name = "The Ultimate Question"
```
Comment (jknpj) [inline code/game/jobs/job/religions.dm]: Are spooks the ultimate question?

Comment (ShiftyRail) [reply code/game/jobs/job/religions.dm]: Whoops
I can't even copypaste properly. Repoban me

File: code/game/jobs/job/religions.dm [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+	deity_names = list("Talos", "Akatosh", "Dibella", "Stendarr", "Kynareth", "Mara", "Arkay", "Julianos", "Zenithar")
+	bible_name = "The Elder Scrolls"
+	male_adept = "Disciple of the Nine"
+	female_adept = "Disciple of the Nine"
+	keys = list("nine divines", "eight divines")
+
+/datum/religion/deadra
+	name = "Cult of the Deadreas"
+	deity_names = list("Azura", "Boethiah", "Sheogorath", "Sanguine", "Hircine", "Meridia", "Hermaeus Mora", "Nocturnal")
+	bible_name = list("The Blessings of Sheogorath", "Boethiah's Pillow Book", "Invocation of Azura")
```
Comment (MadmanMartian) [inline code/game/jobs/job/religions.dm]: Probably out of scope, but Oghma Infinium could be a good alt-title for the bible?

File: code/game/jobs/job/religions.dm [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+	female_adept = "Guru"
+	keys = list("hindu", "hinduism", "india")
+
+/datum/religion/buddism
+	name = "Buddism"
+	deity_name = "Buddha"
+	bible_name = "The Tripitaka"
+	male_adept = "Guru"
+	female_adept = "Guru"
+	keys = list("hindu", "hinduism", "india")
```
Comment (Knightkiller11) [inline code/game/jobs/job/religions.dm]: Key's are duped from hinduism

File: code/game/jobs/job/religions.dm [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+
+/datum/religion/shintoism
+	name = "Shintoism"
+	deity_name = "Kami" //Polytheist and shit, do I sound like a weeb ?
```
Comment (Knightkiller11) [inline code/game/jobs/job/religions.dm]: Weaboo memes aside, gods like Izanagi, Izanami, Susanoo, Amaterasu and Tsukuyomi would be appropriate

File: code/game/jobs/job/religions.dm [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+	female_adept = "Chief Assistant"
+	keys = list("toolboxia", "toolbox")
+
+/datum/religion/homosexuality
+	name = "Homosexuality"
+	deity_name = "Steve Rambo" //Pushing Gaywards
+	bible_names = list("Guys Gone Wild", "Hunk Rump", "It's Okay to be Gay", "Daddy Gave You Good Advice")
+	bible_type = /obj/item/weapon/storage/bible/booze
+	male_adept = "LGBT Advocate"
+	female_adept = "LGBT Advocate"
```
Comment (Knightkiller11) [inline code/game/jobs/job/religions.dm]: Needs keywords added

File: code/game/jobs/job/religions.dm [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+	name = "Science"
+	deity_names = list("Albert Einstein", "Isaac Newton", "Niels Bohr", "Stephen Hawking")
+	bible_names = list("Principle of Relativity", "Quantum Enigma: Physics Encounters Consciousness", "Programming the Universe", "Quantum Physics and Theology", \
+							  "For I Have Tasted The Fruit", "Non-Linear Genetics", "The Mysteries of Bluespace", "Playing God: Collector's Edition")
+	male_adept = "Academician"
+	female_adept = "Academician"
+	keys = list("Science")
+
+/datum/religion/justice
+	name = "Justice"
```
Comment (Knightkiller11) [inline code/game/jobs/job/religions.dm]: Better names would be "Tribunal", "ALMSIVI" or even "Vivec"

File: code/game/jobs/job/religions.dm [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+	name = "Unitology"
+	deity_name = "The Marker"
+	bible_name = "Teachings of Unitology"
+	male_adept = "Vested"
+	female_adept = "Vested"
+	keys = list("unitology", "marker")
+
+/datum/religion/zakarum
+	name = "Zakarum"
+	deity_name = "The Marker"
```
Comment (Knightkiller11) [inline code/game/jobs/job/religions.dm]: Most of this is a copy of Unitology, should be Diablo stuff right?

File: code/game/jobs/job/religions.dm [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+	keys = list("coding", "coder", "coders", "coderbus")
+
+/datum/religion/hitchhiker
+	name = "The Ultimate Question"
+	deity_name = "42"
+	bible_name = "The Hitchhiker's Guide to the Galaxy"
+	bible_type = /obj/item/weapon/storage/bible/booze
+	male_adept = "Hitchhiker"
+	female_adept = "Hitchhiker"
+	keys = list("coding", "coder", "coders", "coderbus")
```
Comment (Knightkiller11) [inline code/game/jobs/job/religions.dm]: Needs different keywords

File: code/game/jobs/job/civilian_chaplain.dm [hunk 10]
```diff
@@ -34,11 +33,41 @@
...
+				if (new_religion == key)
+					rel.misc(H) // We do the misc things related to the religion
+					B = new rel.bible_type
+					B.name = rel.bible_name
+					B.deity_name = rel.deity_name
+					H.put_in_hands(B)
+					if (H.gender == FEMALE)
```
Comment (Kurfursten) [inline code/game/jobs/job/civilian_chaplain.dm]: You can do this with a ternary statement if you care to

Comment (gbasood) [reply code/game/jobs/job/civilian_chaplain.dm]: `J = (H.gender == FEMALE ? rel.female_adept : rel.male_adept)`
not sure the parens would be necessary
it's not a thing you *need* to do or even necessarily *should*, but you *can*

Comment (ShiftyRail) [reply code/game/jobs/job/civilian_chaplain.dm]: I didn't even knew about ternary operators before
I'm always happy to learn

**Comment by ShiftyRail** [comment]
I dislike DM's file tree

File: code/game/jobs/job/religions.dm [hunk 11]
```diff
@@ -563,10 +570,90 @@
...
+	H.equip_or_collect(new /obj/item/clothing/mask/gas/mime(H), slot_wear_mask)
+
+/datum/religion/ancap
+	name = "Anarcho-Capitalism"
+	deity_name = "Murray Rothbard"
+	bible_name = "Bitcoin Wallet"
+	bible_type = /obj/item/weapon/storage/bible/booze
+	male_adept = "Bitcoin Miner" //Worst part coming up with job name
+	female_adept = "Bitcoin Miner"
+	keys = list("silence", "mime", "quiet", "...")
```
Comment (Knightkiller11) [inline code/game/jobs/job/religions.dm]: Keys are duped from mime

File: code/game/jobs/job/religions.dm [hunk 12]
```diff
@@ -563,10 +570,90 @@
...
+	deity_name = "Honkmother"
+	bible_name =  "Honkmothers Coloring Book"
+	male_adept = "Co-Clown"
+	female_adept = "Co-Clown"
+	keys = list("honk", "clown", "honkmother")
+
+/datum/religion/clown/misc(var/mob/living/carbon/human/H)
+	H.equip_or_collect(new /obj/item/clothing/mask/gas/clown_hat(H), slot_wear_mask)
+
+/datum/religion/mime
```
Comment (MadmanMartian) [inline code/game/jobs/job/religions.dm]: This should give the chaplain the silence mutation, if that's within scope

Comment (ShiftyRail) [reply code/game/jobs/job/religions.dm]: @MadmanMartian That would make them strictly speaking mimes with a bible, though

Comment (Arthurdentist) [reply code/game/jobs/job/religions.dm]: It'd actually make them strictly silent mimes with a bible.

**Comment by ShiftyRail** [comment]
Every thing is done and accounted for, ready for merging, unless you have comments

**Comment by ShiftyRail** [comment]
Every thing is done and accounted for, ready for merging, unless you have comments

**Comment by ShiftyRail** [comment]
Fuck

File: code/datums/religions.dm [hunk 13]
```diff
@@ -0,0 +1,661 @@
...
+
+	var/list/keys = list() // What you need to type to get this particular relgion.
+
+/datum/religion/New() // For religions with several bibles/deities
+	if (bible_names.len)
+		bible_name = pick(bible_names)
+	if (deity_names.len)
+		deity_name = pick(deity_names)
+
+/datum/religion/proc/misc(var/mob/living/carbon/human/H, var/obj/item/weapon/storage/bible/B)
```
Comment (PJB3005) [inline code/datums/religions.dm]: Document what this method does with a comment and rename it to something like `equip_chaplain` too. What it does is completely non-obvious without looking at implementations.

File: code/datums/religions.dm [hunk 14]
```diff
@@ -0,0 +1,661 @@
...
+	male_adept = "Stockholder"
+	female_adept = "Stockholder"
+	keys = list("capitalism", "free market", "liberalism", "money")
+
+/datum/religion/capitalism/misc(var/mob/living/carbon/human/H)
+	H.equip_or_collect(new /obj/item/clothing/head/that(H), slot_head)
+
+/datum/religion/america
+	name = "American Exceptionalism"
+	deity_name = "Goerge Washington"
```
Comment (PJB3005) [inline code/datums/religions.dm]: Not sure if typo or intentional.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/datums/religions.dm (hunk 14 above)
Comment (ShiftyRail) [reply code/datums/religions.dm]: This was written like that in the original code

Comment (PJB3005) [reply code/datums/religions.dm]: It definitely was not.

https://github.com/vgstation-coders/vgstation13/blob/bd483ea93c1801b5fd45f38f3d5b8c01fa4cc7ad/code/game/jobs/job/civilian_chaplain.dm#L349

File: code/datums/religions.dm (hunk 13 above)
Comment (ShiftyRail) [reply code/datums/religions.dm]: Did that, also the Bible thing was completely useless and I thought I had it removed several commits ago

File: code/datums/religions.dm [hunk 15]
```diff
@@ -468,19 +469,19 @@
...
 	keys = list("capitalism", "free market", "liberalism", "money")
 
-/datum/religion/capitalism/misc(var/mob/living/carbon/human/H)
+/datum/religion/capitalism/equip_chaplain(var/mob/living/carbon/human/H)
 	H.equip_or_collect(new /obj/item/clothing/head/that(H), slot_head)
 
 /datum/religion/america
 	name = "American Exceptionalism"
-	deity_name = "Goerge Washington"
+	deity_name = "Georges Washington"
```
Comment (PJB3005) [inline code/datums/religions.dm]: Anybody with experience in American history able to confirm whether there were indeed multiple George Washingtons?

Comment (gbasood) [reply code/datums/religions.dm]: I think the question should be, which George's Washington?

Comment (ShiftyRail) [reply code/datums/religions.dm]: Reee
Google told me it was a valid orthography

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**